src/schema/config.label.schema.ts

```diff
@@ -0,0 +1,125 @@
+export interface LabelConfig {
+  // TODO: add additional label config here
+  // e.g., inset label for bar
+
+  // Color & Opacity
+  opacity?: number;
+  fill?: string;
+  fillOpacity?: number;
+  stroke?: string;
+  strokeOpacity?: number;
+  strokeWidth?: number;
+  strokeDash?: number[];
+  strokeDashOffset?: number[];
+
+  // Text-only
+  align?: string;
+  angle?: number;
+  baseline?: string;
+  dx?: number;
+  dy?: number;
+  radius?: number;
+  theta?: number;
+  font?: string;
+  fontSize?: number;
+  fontStyle?: string;
+  fontWeight?: string;
+  // Vega-Lite only for text only
+  format?: string;
+}
+
+export const labelConfig = {
+  type: 'object',
+  properties: {
+    // General Vega
+    fill: {
+      type: 'string',
+      role: 'color',
+      default: '#000000'
+    },
+    opacity: {
+      type: 'number',
+      default: undefined,  // auto
+      minimum: 0,
+      maximum: 1
+    },
+    strokeWidth: {
+      type: 'integer',
+      default: 2,
+      minimum: 0
+    },
+    strokeDash: {
+      type: 'array',
+      default: undefined,
+      description: 'An array of alternating stroke, space lengths for creating dashed or dotted lines.'
+    },
+    strokeDashOffset: {
+      type: 'array',
+      default: undefined,
+      description: 'The offset (in pixels) into which to begin drawing with the stroke dash array.'
+    },
+    // text-only
+    align: {
+      type: 'string',
+      default: 'right',
```

Comment:
@willium you might wanna change this default to `undefined` since you have a rule for determining it.